spec.emu

```diff
@@ -0,0 +1,1319 @@
+
+        <p>A fix to this problem is being worked on, and is expected to be included in a future version of the standard. It will likely involve early returning from the below algorithms whenever there is a comment (or comment-like) that contains the characters U+0060 (&#x60;), U+0022 ("), or U+0027 ('), or the the sequence U+002A U+002F (*/).</p>
+      </emu-note>
+
+      <emu-clause id="sec-JavaScriptExtractSourceMapURL" type="abstract operation">
+        <h1>
+          JavaScriptExtractSourceMapURL (
+            _source_: a String,
+          ): a String or *null*
+        </h1>
+        <dl class="header">
+          <dt>description</dt>
+          <dd>It extracts a source map URL from a <strong>JavaScript</strong> source. It has two possible implementations: either <emu-xref href="#sec-JavaScriptExtractSourceMapURL-through-parsing">through parsing</emu-xref> or without parsing.</dd>
+        </dl>
+
+        <p>To extract a source map URL <dfn id="sec-JavaScriptExtractSourceMapURL-through-parsing">through parsing</dfn>:</p>
+
+        <emu-alg>
+          1. Let _tokens_ be the List of tokens obtained by parsing _source_ according to <emu-xref href="#sec-ecmascript-language-lexical-grammar">ECMA-262's lexical grammar</emu-xref>.
+          1. For each _token_ in _tokens_, in reverse order, do:
+            1. If _token_ is not |SingleLineComment| or |MultiLineComment|, return *null*.
+            <!-- TODO: We need to define a SDO for this, to make explicit what the contents actually are -->
+            1. Let _comment_ be the content of _token_.
+            1. Let _sourceMapURL_ be MatchSourceMapURL(_comment_).
+            1. If _sourceMapURL_ is a String, return _sourceMapURL_.
+          1. Return *null*.
+        </emu-alg>
+
+        <p>To extract a source map URL <dfn id="sec-JavaScriptExtractSourceMapURL-without-parsing">without parsing</dfn>:</p>
+
+        <emu-alg>
+          1. Let _lines_ be CreateListFromArrayLike(Call(%String.prototype.split%, _mappings_, « RegExpCreate(*"\u000D\u000A|\u000A|\u000D|\u2028|\u2029"*, *"g"*) »)).
+          1. NOTE: The regular expression above matches the |LineTerminatorSequence| production.
+          1. Let _lastURL_ be *null*.
+          1. For each _lineStr_ in _lines_, do:
+            1. Let _line_ be StringToCodePoints(_lineStr_).
+            1. Let _position_ be 0.
+            1. Let _lineLength_ be the length of _line_.
+            1. While _position_ &lt; _lineLength_,
+              1. Let _first_ be _line_[_position_].
+              1. Set _position_ to _position_ + 1.
+              1. If _first_ is U+002F (SOLIDUS) and _position_ &lt; _lineLength_, then
```

Comment:
I noticed sometimes U+002F is SOLIDUS like here and sometimes /. Should we make it always consistently SOLIDUS?